cylc/flow/parsec/fileparse.py

```diff
@@ -224,6 +229,27 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     do_empy = True
     do_jinja2 = True
     do_contin = True
+
+    if not template_vars:
+        template_vars = {}
+
+    # Load Rose Vars, if a ``rose-suite.conf`` file is present.
+    extra_vars = {
+        'env': {},
+        'template variables': {},
+        'templating detected': None
+    }
+    for entry_point in pkg_resources.iter_entry_points(
+        'cylc.pre_configure'
+    ):
+        plugin_result = entry_point.resolve()(Path(fpath).parent)
+        for section in ['env', 'template variables']:
+            if plugin_result[section] is not None:
+                extra_vars[section].update(plugin_result.get(section, {}))
```

Comment:
This is potentially problematic in case of multiple plugins trying to write the same variables (e.g., `$PATH`), cf. https://github.com/cylc/cylc-flow/pull/3913#discussion_r530333657.

Reply:
I'm not sure I have a quick or clever answer...

@oliver-sanders ?

Reply:
I put a comment on the other thread - https://github.com/cylc/cylc-flow/pull/3913#discussion_r531685998

I think if two plugins try to write the same variables then the flow design is flawed. I don't think we want to build a system to handle inter-dependence between installation plugins. Easier just to say, don't do that.

Reply:
> I think if two plugins try to write the same variables then the flow design is flawed. I don't think we want to build a system to handle inter-dependence between installation plugins. Easier just to say, don't do that.

I think it's best to detect collisions and throw an error in that case:

```suggestion
                section_update = plugin_result.get(section, {})
                keys_collision = extra_vars[section].keys() & section_update.keys()
                if keys_collision:
                    raise ParsecError(section + " collision: " + ", ".join(sorted(keys_collision)))
                extra_vars[section].update(section_update)
```